tachyon/crypto/commitments/merkle_tree/field_merkle_tree/field_merkle_tree.h

```diff
@@ -116,7 +116,9 @@ class FieldMerkleTree {
     return {std::move(leaves), std::move(digest_layers)};
   }
 
-  const std::vector<math::RowMajorMatrix<F>>& leaves() const { return leaves_; }
+  const std::vector<Eigen::Map<math::RowMajorMatrix<F>>>& leaves() const {
+    return leaves_;
+  }
```

Comment:
```suggestion
  const std::vector<Eigen::Map<const math::RowMajorMatrix<F>>>& leaves() const {
    return leaves_;
  }
```

Otherwise, I think this is not really a `const` function.

Reply:
1.` const Eigen::Map<math::RowMajorMatrix<F>>` will expose the methods that uses `const F*`. So I think it's fine. See https://gitlab.com/libeigen/eigen/-/blob/84282c42fc05d08216331203206de7ceb024a9c6/Eigen/src/Core/DenseCoeffsBase.h#L102-109 and https://gitlab.com/libeigen/eigen/-/blob/84282c42fc05d08216331203206de7ceb024a9c6/Eigen/src/Core/DenseCoeffsBase.h#L44-53.
2. Since `std::vector<Eigen::Map<math::RowMajorMatrix<F>>>` and `std::vector<Eigen::Map<const math::RowMajorMatrix<F>>>` are completely different type, I can't just modify the type in this way!